mindmeld/app.py

```diff
@@ -171,6 +172,20 @@ def _decorator(func):
 
         return _decorator
 
+    def auto_fill(self, **kwargs):
+        """Creates a flow to fill missing entities"""
+
```

Comment:
Can we add a `functools.wraps()` here to keep the signature and docstrings of the wrapped function?

Reply:
Following from that, unless you need to pass keywords through without modification, it's better to provide explicitly named parameters with defaults rather than `kwargs`.

```python
def auto_fill(self, name=None, entity_form=None, max_retries=1):
    ...
```

Reply:
Following from that, you may want something that looks like

`
def auto_fill(self, name=None, max_retries=1, *, entity_form, **kwargs):
    ...`

- You need `**kwargs` to pass it into `add_dialogue_rule`
- Making `entity_form` a required keyword-only arg performs the assertion in line 181 at parse time rather than runtime.  Hence you can drop that assertion.

Reply:
I guess since this is from the developer point of view, shouldn't the assertion be performed at parse time and fail the app from loading? If that assertion is removed, the app is allowed to load even with a empty form and throw an error at runtime (user-facing).

Reply:
Let's keep the assertion but change the function signature here to be more explicit.